x-pack/plugins/security_solution/public/management/pages/policy/view/host_isolation_exceptions/components/assign_flyout.tsx

```diff
@@ -0,0 +1,303 @@
+            }
+          )}
+        </EuiCallOut>
+        <EuiSpacer size="m" />
+      </>
+    ),
+    []
+  );
+
+  const noItemsMessage = useMemo(() => {
+    if (exceptionsRequest.isLoading || allPossibleExceptionsRequest.isLoading) {
+      return null;
+    }
+
+    // there are no host isolation exceptions assignable to this policy
+    if (
+      allPossibleExceptionsRequest.data?.total === 0 ||
+      (exceptionsRequest.data?.total === 0 && currentFilter === '')
+    ) {
+      return (
+        <EuiEmptyPrompt
+          data-test-subj="hostIsolationExceptions-no-assignable-items"
+          title={
+            <FormattedMessage
+              id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.noAssignable"
+              defaultMessage="There are no host isolation exceptions that can be assigned to this policy."
+            />
+          }
+        />
+      );
+    }
+
+    // there are no results for the current search
+    if (exceptionsRequest.data?.total === 0) {
+      return (
+        <EuiEmptyPrompt
+          data-test-subj="hostIsolationExceptions-no-items-found"
+          title={
+            <FormattedMessage
+              id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.noResults"
+              defaultMessage="No items found"
+            />
+          }
+        />
+      );
+    }
+  }, [
+    allPossibleExceptionsRequest.data?.total,
+    allPossibleExceptionsRequest.isLoading,
+    currentFilter,
+    exceptionsRequest.data?.total,
+    exceptionsRequest.isLoading,
+  ]);
+
+  return (
+    <EuiFlyout onClose={onClose} data-test-subj="hostIsolationExceptions-assign-flyout">
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle size="m">
+          <h2>
+            <FormattedMessage
+              id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.title"
+              defaultMessage="Assign host isolation exceptions"
+            />
+          </h2>
+        </EuiTitle>
+        <EuiSpacer size="m" />
+        <FormattedMessage
+          id="xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.flyout.subtitle"
+          defaultMessage="Select host isolation exceptions to add to {policyName}"
+          values={{ policyName: policy.name }}
+        />
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        {(exceptionsRequest.data?.total || 0) > MAX_ALLOWED_RESULTS ? searchWarningMessage : null}
+        <SearchExceptions
+          onSearch={handleOnSearch}
+          placeholder={i18n.translate(
+            'xpack.securitySolution.endpoint.policy.hostIsolationExceptions.layout.searh.label',
+            {
+              defaultMessage: 'Search host isolation exceptions',
+            }
+          )}
+          hideRefreshButton
+        />
+        <EuiSpacer size="m" />
+
+        {!exceptionsRequest.isLoading ? (
```

Comment:
Is this correct? The `PolicyArtifactsAssignableList` has a prop `isListLoading` that displays a loader in the list, should you rely in that prop and not hide the whole component when it's loading?

Reply:
True! I'll remove this wrapper.